site/_posts/2019-09-20-Velero-v1-1-on-PKS.md

```diff
@@ -0,0 +1,350 @@
+
+* Note: Stateful backups of applications deployed in Kubernetes on vSphere that use the `restic` plugin for backing up Persistent Volumes would send the backup data to the same S3 Object Store.
+
+There are a few different steps required to successfully deploy the Minio S3 Object Store.
+
+### 1. Create a Minio credentials secret file
+
+A simple credentials file containing the login/password (id/key) for the local on-premises Minio S3 Object Store must be created.
+
+```bash
+$ cat credentials-velero
+[default]
+aws_access_key_id = minio
+aws_secret_access_key = minio123
+```
+
+### 2. Expose Minio Service on a NodePort
+
+While this step is optional, it is useful for two reasons. The first is that it gives you a way to access the Minio portal through a browser and examine the backups. The second is that it enables you to specify a `publicUrl` for Minio, which in turn means that you can access backup and restore logs from the Minio S3 Object Store.
+
+To expose the Minio Service on a NodePort, a modification of the `examples/minio/00-minio-deployment.yaml` manifest is necessary. The only change is to the type: field, from ClusterIP to NodePort:
+
+```bash
+spec:
+# ClusterIP is recommended for production environments.
+# Change to NodePort if needed per documentation,
+# but only if you run Minio in a test/trial environment, for example with Minikube.
+type: NodePort
+```
+
+### 3. Create the Minio Object Store
+
+After making the changes above, simply run the following command to create the Minio Object Store.
+
+```bash
+$ kubectl apply -f examples/minio/00-minio-deployment.yaml
+namespace/velero created
+deployment.apps/minio created
+service/minio created
+job.batch/minio-setup created
+```
+
+### 4. Verify Minio Object Store has deployed successfully
+
+Retrieve both the Kubernetes node on which the Minio Pod is running, and the port that the Minio Service has been exposed on. With this information, you can verify that Minio is working.
+
+```bash
+$ kubectl get pods -n velero
+NAME                     READY   STATUS      RESTARTS   AGE
+minio-66dc75bb8d-95xpp   1/1     Running     0          25s
+minio-setup-zpnfl        0/1     Completed   0          25s
+```
+
+```bash
+$ kubectl describe pod minio-66dc75bb8d-95xpp -n velero | grep -i Node:
+Node:               140ab5aa-0159-4612-b68c-df39dbea2245/192.168.192.5
+```
+
+```bash
+$ kubectl get svc -n velero
+NAME    TYPE       CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
+minio   NodePort   10.100.200.82   <none>        9000:32109/TCP   5s
+```
+
+In the above outputs, the node on which the Minio Object Storage is deployed has IP address `192.168.192.5`. The NodePort that the Minio Service is exposed is `32109`. If we now direct a browser to that `Node:port` combination, we should see the Minio Object Store web interface. You can use the credentials provided in the `credentials-velero` file earlier to login.
+
+![Minio Object Store](../img/vsphere-tutorial-icons/Minio.png)
```

Comment:
this image ref is broken in the site preview - looks like the file doesn't exist.

Reply:
Good catch - forgot to include the folder when we re-created the PR.